engine/access/rpc/backend/backend_scripts.go

```diff
@@ -107,7 +113,9 @@ func (b *backendScripts) executeScriptOnExecutionNode(
 		if err == nil {
 			if b.log.GetLevel() == zerolog.DebugLevel {
 				executionTime := time.Now()
-				timestamp, seen := b.seenScripts[encodedScript]
+				b.seenScripts.lock.RLock()
+				timestamp, seen := b.seenScripts.scripts[encodedScript]
+				b.seenScripts.lock.RUnlock()
```

Comment:
It's better to wrap this into a Getter and Setter method, in case somewhere else using it and forgot to wrap